### DISCUSSION TITLE
[FEAT]  add admin actions chart

### DISCUSSION THREAD

**Discussion by erfjab** [marzneshin/marzneshin#729] [Ideas]
### Description

درود و وقت بخیر،  

یکی از قابلیت‌های مفید برای پنل اینه که بتونیم فعالیت‌های ادمین‌ها رو به صورت آماری ببینیم. مثلا اینکه هر ادمین چقدر کاربر اضافه یا حذف کرده و چه مقدار ترافیک مصرف شده. پیشنهاد می‌کنم یه بخش جدید به داشبورد اضافه بشه که آمار فعالیت ادمین‌ها رو به صورت زیر نشون بده:

```
| Admin  | Users Add | Users Delete | Users Modify | Traffic  |
|--------|-----------|-------|-------|----------|
| Jafar  | +45       | -12      | 1   | +785 Gb  |
| Gholam | +78       | -102     | 8    | +705 Gb  |
```

### جزئیات:
- **Users Add**: تعداد کاربرانی که ادمین ایجاد کرده.
- **Users Delete**: تعداد کاربرانی که ادمین ویرایش کرده
- **Users Delete**: تعداد کاربرانی که ادمین حذف کرده.
- **Traffic**: ترافیک مصرفی کاربران مربوط به هر ادمین.

### Suggested Solution

برای این مورد، میشه API رو ارتقا داد و از `state admins` استفاده کرد تا این اطلاعات رو از طریق دیتابیس به نمایش گذاشت.
برای مورد ساخت یوزر و ترافیک مصرفی مشکلی نداریم
برای مدیفای یوزر اطمینان کافی ندارم که edit at برای ادیت های سیستم هم ثبت میشه یا نه
برای حذف یوزر هم باید یه فیلد جدید به اسم removed at استفاده کنیم تا حل بشه

**Comment by Kiya6955** [comment]
> ### Description
> 
> درود و وقت بخیر،  
> 
> 
> 
> یکی از قابلیت‌های مفید برای پنل اینه که بتونیم فعالیت‌های ادمین‌ها رو به صورت آماری ببینیم. مثلا اینکه هر ادمین چقدر کاربر اضافه یا حذف کرده و چه مقدار ترافیک مصرف شده. پیشنهاد می‌کنم یه بخش جدید به داشبورد اضافه بشه که آمار فعالیت ادمین‌ها رو به صورت زیر نشون بده:
> 
> 
> 
> ```
> 
> | Admin  | Users Add | Users Delete | Users Modify | Traffic  |
> 
> |--------|-----------|-------|-------|----------|
> 
> | Jafar  | +45       | -12      | 1   | +785 Gb  |
> 
> | Gholam | +78       | -102     | 8    | +705 Gb  |
> 
> ```
> 
> 
> 
> ### جزئیات:
> 
> - **Users Add**: تعداد کاربرانی که ادمین ایجاد کرده.
> 
> - **Users Delete**: تعداد کاربرانی که ادمین ویرایش کرده
> 
> - **Users Delete**: تعداد کاربرانی که ادمین حذف کرده.
> 
> - **Traffic**: ترافیک مصرفی کاربران مربوط به هر ادمین.
> 
> 
> 
> ### Suggested Solution
> 
> برای این مورد، میشه API رو ارتقا داد و از `state admins` استفاده کرد تا این اطلاعات رو از طریق دیتابیس به نمایش گذاشت.
> 
> برای مورد ساخت یوزر و ترافیک مصرفی مشکلی نداریم
> 
> برای مدیفای یوزر اطمینان کافی ندارم که edit at برای ادیت های سیستم هم ثبت میشه یا نه
> 
> برای حذف یوزر هم باید یه فیلد جدید به اسم removed at استفاده کنیم تا حل بشه
> 
> ### Alternatives
> 
> _No response_
> 
> ### Additional Context
> 
> _No response_

به این میگن تجارت

**Comment by Imnonamee** [comment]
# **عالی و کاربردی**
* تجارت موفق

**Comment by erfjab** [comment]
خب ایشو دیسکاشن شد، خوب شد.

**Comment by AsanFillter** [comment]
این قابلیت واقعاً یکی از نکات کلیدی و بسیار کاربردی هست به خصوص برای افرادی که به عنوان ادمین پنل رو به فرض مثال دست دوست و آشنا میدن یا اشتراک ادمینی به فروش میرسونن میتونه کمک کنه تا دید جامع‌تر و دقیقی نسبت به عملکرد کاربران و ادمین‌های خودشون داشته باشن و مدیریت بهتری انجام بدن.

**Comment by alimahjoub** [comment]
منم موافقم این قابلیت ها اضافه بشه برای نمایندگی دادن خیلی بهتر میشه

**Comment by gasparno3** [comment]
دکمه Reset Users Data Usage برای ادمین ها اضافه بشه خوب هست
یا دیدن مصرف ماهانه یا بازه زمانی خاص یک ادمین

**Reply by alimahjoub** [reply]
به نظر من دومی بهتره.
هر چند الان مصرف کلی ادمین نشون داده میشه قسمت تنظیمات ادمین میشه بر اساس اونجا با ادمین حساب کتاب کنید
